Detectors/MUON/MCH/Simulation/src/Response.cxx

```diff
@@ -90,20 +89,57 @@ uint32_t Response::nSamples(float charge) const
+      if ((betagamma > 3.2) && (thetawire * TMath::RadToDeg() <= 15.)) {
+        betagamma = log(betagamma); // check if ln or log10
+        eLossParticleElossMip = eLossRatio(betagamma);
+        sigmaEffect10degrees = angleEffect10(eLossParticleElossMip);
+        sigmaEffectThetadegrees = sigmaEffect10degrees / angleEffectNorma(thetawire * TMath::RadToDeg());
+        if (o2::mch::Station() == o2::mch::Station::Type1) {
+          sigmaEffectThetadegrees /= 1.09833 + 0.017 * (thetawire * TMath::RadToDeg());
+        }
+        yAngleEffect = 0.0001 * gRandom->Gaus(0, sigmaEffectThetadegrees); // error due to the angle effect in cm
+      }
+    } else {
+      if ((betagamma > 3.2) && (abs(thetawire * TMath::RadToDeg()) <= 15.)) {
+        betagamma = log(betagamma);
+        eLossParticleElossMip = eLossRatio(betagamma);
+        sigmaEffect10degrees = angleEffect10(eLossParticleElossMip);
+        sigmaEffectThetadegrees = sigmaEffect10degrees / magAngleEffectNorma(thetawire * TMath::RadToDeg(), bx / 10.); // check b-field unit in aliroot and O2
+        if (o2::mch::Station() == o2::mch::Station::Type1) {
+          sigmaEffectThetadegrees /= 1.09833 + 0.017 * (thetawire * TMath::RadToDeg());
+        }
+        yAngleEffect = 0.0001 * gRandom->Gaus(0, sigmaEffectThetadegrees);
+      }
+    }
+  }
+  return yAngleEffect;
+}
+//_____________________________________________________________________
 float Response::eLossRatio(float logbetagamma) const
 {
   // Ratio of particle mean eloss with respect MIP's Khalil Boudjemline, sep 2003, PhD.Thesis and Particle Data Book
   /// copied from aliroot AliMUONv1.cxx
   float eLossRatioParam[5] = {1.02138, -9.54149e-02, +7.83433e-02, -9.98208e-03, +3.83279e-04};
-  return eLossRatioParam[0] + eLossRatioParam[1] * logbetagamma + eLossRatioParam[2] * std::pow(logbetagamma, 2) + eLossRatioParam[3] * std::pow(logbetagamma, 3) + eLossRatioParam[4] * std::pow(logbetagamma, 4);
+  return eLossRatioParam[0] + eLossRatioParam[1] * logbetagamma + eLossRatioParam[2] * logbetagamma * logbetagamma + eLossRatioParam[3] * logbetagamma * logbetagamma * logbetagamma + eLossRatioParam[4] * logbetagamma * logbetagamma * logbetagamma * logbetagamma;
 }
 //_____________________________________________________________________
 float Response::angleEffect10(float elossratio) const
 {
   /// Angle effect in tracking chambers at theta =10 degres as a function of ElossRatio (Khalil BOUDJEMLINE sep 2003 Ph.D Thesis) (in micrometers)
   /// copied from aliroot AliMUONv1.cxx
   float angleEffectParam[3] = {1.90691e+02, -6.62258e+01, 1.28247e+01};
-  return angleEffectParam[0] + angleEffectParam[1] * elossratio + angleEffectParam[2] * std::pow(elossratio, 2);
+  return angleEffectParam[0] + angleEffectParam[1] * elossratio + angleEffectParam[2] * elossratio * elossratio;
 }
 //_____________________________________________________________________
 float Response::angleEffectNorma(float elossratio) const
```

Comment:
for the sake of clarity, the name of the parameter should be `angle` instead of `elossratio`

Reply:
ok.